### DISCUSSION TITLE
🐦🐦征集新图标 Call for new icon 🐦🐦

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#370] [General]
Magpie 接下来将迁移到 Win11 风格的用户界面，在此征集新图标👋👋。贡献者会加入鸣谢清单。

要求：Fluent Design 风格，在黑色和白色背景下以及尺寸较小时（系统托盘图标大小）有较高辨识度

**Comment by XiaofengdiZhu** [comment]
来自[icons8](https://icons8.com/icon/set/magni/fluency)的免费图标（注意：Link to Icons8 when you use our assets.）
![](https://img.icons8.com/fluency/240/FFFFFF/search.png)

**Reply by Blinue** [reply]
谢谢。这个图标和 Fluent Design 风格差别挺大（不必局限于放大镜）

**Reply by XiaofengdiZhu** [reply]
> 谢谢。这个图标和 Fluent Design 风格差别挺大（不必局限于放大镜）

em，没仔细研究过……个人以为渐变色块+亚克力半透明就是fluent

**Reply by Blinue** [reply]
https://docs.microsoft.com/zh-cn/windows/apps/design/style/iconography/app-icon-design

**Reply by rxia24** [reply]
稍微改了下:

![Bird](https://user-images.githubusercontent.com/22437401/195056950-06ea9326-ac6b-4128-abf8-a83e6f2699d4.png)


原图来自：[edt.im](https://www.flaticon.com/free-icon/bird_8164019?term=bird&page=1&position=95&page=1&position=95&related_id=8164019&origin=style)

**Comment by Blinue** [comment]
> 稍微改了下:
> 
> ![Bird](https://user-images.githubusercontent.com/22437401/195056950-06ea9326-ac6b-4128-abf8-a83e6f2699d4.png)
> 
> 原图来自：[edt.im](https://www.flaticon.com/free-icon/bird_8164019?term=bird&page=1&position=95&page=1&position=95&related_id=8164019&origin=style)

@rxia24 非常感谢。我在考虑用鸟做图标是否妥当，这个图标无法体现 Magpie 的功能。关于 Magpie 名称的由来 https://github.com/Blinue/Magpie/issues/94#issuecomment-968756472

> An icon should be a metaphor for its app: a visual representation of the value proposition, functions, and features of the product.

**Comment by Blinue** [comment]
![48](https://user-images.githubusercontent.com/34770031/196181651-02908960-8634-4e6f-8899-b203074dde84.png)
姑且搞了一个，改自 PowerToys。先凑合用

**Reply by WluhWluh** [reply]
如果要继续使用这个图标我可以重画一下提高分辨率，也可以看看我新发的大致方案。

**Reply by WluhWluh** [reply]
![画板 21@8x (自定义)](https://user-images.githubusercontent.com/52004526/233265215-e8ebbaf3-88ef-4b18-b7fe-d5ce77117734.png)

**Comment by WluhWluh** [comment]
![画板 4@8x (自定义)](https://user-images.githubusercontent.com/52004526/233133591-5fd7168f-5818-4746-89b8-e09425dcae9e.png)
采用喜鹊辨识度最高的飞羽图案变形放在右侧，左侧是窗口及代表缩放的扇形箭头。

**Reply by WluhWluh** [reply]
![画板 4@8x无阴影 (自定义)](https://user-images.githubusercontent.com/52004526/233136158-7dca8000-2c8a-4a09-a270-43936ea55f3c.png)
无外部阴影版本

**Comment by WluhWluh** [comment]
![画板 5@8x (自定义)](https://user-images.githubusercontent.com/52004526/233143756-41a1cf86-5b1b-4538-a30d-140dd50d068c.png)
可以在托盘使用的简洁小图标

**Reply by WluhWluh** [reply]
![画板 5@8x1 (自定义)](https://user-images.githubusercontent.com/52004526/233145740-c184f47b-e41d-46a1-b2d5-2b3fe57edbe4.png)

**Reply by WluhWluh** [reply]
![画板 6@8x1 (自定义)](https://user-images.githubusercontent.com/52004526/233147988-5d5b165e-832c-405e-927c-91c9c3e1439e.png)

**Comment by WluhWluh** [comment]
![画板 8@8x6 (自定义)](https://user-images.githubusercontent.com/52004526/233184572-108a4a10-a120-47f9-be27-3f26500da29f.png)
![magpie预览 拷贝](https://user-images.githubusercontent.com/52004526/233184621-10f3fda5-30cc-4f11-9d46-c6851d66a339.jpg)
感觉我之前的方案还是太杂，再精简一下效果好些

**Reply by Blinue** [reply]
你设计的这几个都很好，非常感谢！我个人倾向于不规则轮廓和有活力的配色，如果你有其他想法也请画出原型图，最后我会搞一个投票决定。

**Reply by WluhWluh** [reply]
好的，我再做几版方案

**Comment by WluhWluh** [comment]
![画板 2](https://user-images.githubusercontent.com/52004526/233248497-d2607cfb-5c26-40f0-bb59-c483543a3942.png)

**Reply by Blinue** [reply]
虽然好看但元素太多了，放在托盘菜单上可能看不清。在 100% 缩放下托盘图标大小为 32x32，只能容纳一到两个元素。（也可以专门为托盘图标制作简化版本）

**Comment by WluhWluh** [comment]
![羽毛@8x (自定义)](https://user-images.githubusercontent.com/52004526/233252639-43bc6ae0-25df-474e-a163-d52f6a16a5bf.png)
![屏幕截图 2023-04-20 113630 (小)](https://user-images.githubusercontent.com/52004526/233252663-b95cc500-356d-4fe3-9122-d60bd94c358e.jpg)

**Reply by Blinue** [reply]
太棒了，虽然不能展示功能，但很简洁美观，尺寸很小也有辨识度。

或许可以大尺寸图标使用前面的某一个，小尺寸图标使用这个？

**Reply by Blinue** [reply]
一个小提醒：在图标的四周要留一些空间（plated icon），否则在显示时不协调

![图片](https://user-images.githubusercontent.com/34770031/233256483-dc3b522e-7286-4638-8afc-8ebd060e5132.png)

**Reply by WluhWluh** [reply]
感谢提醒

**Reply by WluhWluh** [reply]
![画板 6@8x6 (自定义)](https://user-images.githubusercontent.com/52004526/233264632-252bb40f-281f-4f48-a113-ce7b5d1e0ed1.png)
![资源 5@8x234 (自定义)](https://user-images.githubusercontent.com/52004526/233264725-171b7b67-3ccb-4caf-ac44-cb8ede58e009.png)
![新预览](https://user-images.githubusercontent.com/52004526/233264754-4fa73938-69f1-46fa-994e-ac470a9914fd.png)
修改了羽干的形状

**Reply by Blinue** [reply]
@hooke007 你觉得怎么样？

**Reply by hooke007** [reply]
我觉得可以融合一下 https://github.com/Blinue/Magpie/discussions/370#discussioncomment-5666042 里左上的点，mapie在激活缩放模式的时候点亮/变蓝点，不使用缩放模式的时候恢复原色（动态图标）

**Reply by WluhWluh** [reply]
![画板 4 副本@8x7 (自定义)](https://user-images.githubusercontent.com/52004526/233270727-7620b9d2-ebb8-4275-a889-ae414654e845.png)
![画板 4 副本@8x9 (自定义)](https://user-images.githubusercontent.com/52004526/233270753-5c6288af-3795-4c65-9620-58394ab71ecd.png)
![画板 6@8x8 (自定义)](https://user-images.githubusercontent.com/52004526/233270788-52ee09ca-75e4-47ea-97fd-697cd5b9b427.png)
点和羽毛变色的尝试，大概是这个意思吗？

**Reply by hooke007** [reply]
是的

**Reply by Blinue** [reply]
> 我觉得可以融合一下 https://github.com/Blinue/Magpie/discussions/370#discussioncomment-5666042 里左上的点，mapie在激活缩放模式的时候点亮/变蓝点，不使用缩放模式的时候恢复原色（动态图标）

大多数人只有一个屏幕，缩放时看不到 Magpie 的图标变化。另外我觉得羽毛变色后不太协调

**Reply by hooke007** [reply]
我想的比较长远，窗口化缩放

**Reply by Blinue** [reply]
支持窗口化的话这个确实有用。处于缩放状态下右上角的点变为蓝色，但羽毛颜色不变
@WluhWluh 能麻烦你制作这样两个图标吗？新图标很棒，我认为没有投票的必要了。请直接提一个 PR。

**Reply by Blinue** [reply]
需要的图标尺寸：https://learn.microsoft.com/zh-cn/windows/apps/design/style/iconography/app-icon-construction#icon-scaling

至少要支持 24x24、32x32、48x48 和 256x256

**Reply by WluhWluh** [reply]
好的，我做一下多尺寸。

**Reply by Blinue** [reply]
![图片](https://user-images.githubusercontent.com/34770031/233278192-e3c29bb9-94d1-43bd-8085-75355815fec0.png)
标题栏依然不协调，请把四周间距再加大一些，比例参考：
![图片](https://user-images.githubusercontent.com/34770031/233278567-996ea455-ee1b-4aea-8286-9e36287a50d5.png)

**Reply by hooke007** [reply]
啊等等，这个怎么那么像tcltk
或者说python自带的tkinter

**Reply by WluhWluh** [reply]
制作的时候参考了Windows11 emoji的羽毛外观
![feather_1fab6](https://user-images.githubusercontent.com/52004526/233283840-42a9b866-eafe-44a2-a001-b4578a1ad218.png)
和tcltk还有tkinker主要像在配色上，问题应该不大吧？

**Reply by Blinue** [reply]
![tcltk](https://user-images.githubusercontent.com/34770031/233282894-ff4794df-3421-4e6f-8563-8aeaec8f522a.png)

![tkinter](https://programacionfacil.org/images/cursos/tkinter/xtkinter-logo.png.pagespeed.ic.o56GLgFeDe.png)

只有小尺寸才只有一个羽毛，我认为样式区别挺大的。如果要避嫌的话换个颜色怎么样？

**Reply by hooke007** [reply]
不是，就是既视感有点强，特别是作为后台图标，我一眼有点看叉了
问题不大
![Snipaste_2023-04-20_07-49-46](https://user-images.githubusercontent.com/41094733/233284491-732d009d-e964-4096-9bf3-c67da9fcbea1.jpg)
![Snipaste_2023-04-20_07-51-58](https://user-images.githubusercontent.com/41094733/233284494-80aa4c01-2c20-4b79-932d-9127ade668fd.jpg)

**Reply by WluhWluh** [reply]
@Blinue 前面发的实际应用预览图全都是完全没加边距的
![边距测试1](https://user-images.githubusercontent.com/52004526/233284426-47fcca03-bb7a-4014-9d66-2a1ae8c0cdbb.png)
目前考虑左右各加1/16总宽度的边距，感觉加多了图标可能过小。加边距后由上图右侧缩至上图左侧大小

**Reply by Blinue** [reply]
> 目前考虑左右各加1/16总宽度的边距，感觉加多了图标可能过小。加边距后由上图右侧缩至上图左侧大小

和Magpie目前的图标对比怎么样？

**Reply by WluhWluh** [reply]
![屏幕截图(599) 拷贝](https://user-images.githubusercontent.com/52004526/233287433-f2aa50ea-01ae-4f36-abeb-5b271050c4db.jpg)

**Reply by Blinue** [reply]
别加了，我看官方软件似乎都没有边距。打包应用可以分别提供有边距和无边距的版本，非打包应用的机制我不清楚。

**Reply by WluhWluh** [reply]
好的

**Reply by WluhWluh** [reply]
![image](https://user-images.githubusercontent.com/52004526/233288604-b2d56318-55a6-43d1-877a-302d7403fe05.png)
羽毛颜色还需要更改吗？或者在蓝色系微调？

**Reply by Blinue** [reply]
我认为不用改，Windows 大多数原生应用的图标都是蓝色的。如果不是 hooke007 的提醒，我甚至没听说过 tcltk...

**Reply by hooke007** [reply]
> 我甚至没听说过 tcltk...

可能过时了，而且也没什么人用 xwindow上打不过gtk,qt ；python-gui里tkinter也打不过pyqt

**Reply by WluhWluh** [reply]
图标都上传到新的Icon文件夹里了，没有动老图标。每种变体都提供了不同尺寸和svg矢量图。没有在同一ico文件里组合不同变体。
